doc/source/data/advanced-pipelines.rst

```diff
@@ -136,7 +136,7 @@ Readers pulling batches from the pipeline will see the same data blocks repeated
 Pre-repeat vs post-repeat transforms
 ====================================
 
-Transformations made prior to the Dataset prior to the call to ``.repeat()`` are executed once. Transformations made to the DatasetPipeline after the repeat will be executed once for each repetition of the Dataset.
+Transformations made prior to the Dataset prior to the call to ``.repeat()`` may be re-used. Transformations made to the DatasetPipeline after the repeat will always be executed once for each repetition of the Dataset.
```

Comment:
Minor, but "may be re-used" kind of sounds like the system will decide for the user. Maybe make it clearer how/when to opt in?